lib/blob.js

```diff
@@ -147,17 +240,164 @@ function Blob(options) {
   if (this.options.accessKey) {
     // If set authorize to use shared key signatures
     this.authorize = authorizeWithSharedKey;
+
     // Decode accessKey
     this._accessKey = new Buffer(this.options.accessKey, 'base64');
+  } else if (this.options.sas instanceof Function) {
+    // Set authorize to use shared-access-signatures with refresh function
+    this.authorize = authorizeWithRefreshSAS;
+    // Set state with _nextSASRefresh = -1, we'll refresh on the first request
+    this._nextSASRefresh = -1;
+    this._sas = '';
+  } else if (typeof(this.options.sas) === 'string') {
+    // Set authorize to use shared-access-signature as hardcoded
+    this.authorize = authorizeWithSAS;
   } else {
-    throw new Error("Either options.accessKey, options.sas as function or " +
-      "options.sas as string must be given!");
+    this.authorize = anonymous;
   }
+
+  // Initialize EventEmitter parent class
+  events.EventEmitter.call(this);
 };
 
 // Export Blob
 module.exports = Blob;
 
+// Subclass EventEmitter
+util.inherits(Blob, events.EventEmitter);
+
+/**
+ * Generate a SAS string on the form 'key1=va1&key2=val2&...'.
+ *
+ * @param {string}  container - Name of the container that this SAS string applies to.
+ * @param {string}  blob - Name of the blob that this SAS string applies to.
+ * @param {object} options - Options for the following form:
+ *  * ```js
+ * {
+ *   start:               new Date(), // Time from which signature is valid (optional)
+ *   expiry:              new Date(), // Expiration of signature (required). Can be omitted if it is specified in the
+ *                                    // associated stored access policy
+ *   resourceType:        '...',      // Specifies which resources are accessible via the SAS. (required)
```

Comment:
my thoughts on default values is that if it requires more than a split second to figure out which should be the default, there shouldn't be a default.  In this case, I can't really pick one or the other, so I'd lean towards not having a default